### PR TITLE
feat(input): add input-stack molecule

### LINKED ISSUE #396: Support inline and block style form elements
Currently form elements such as text input fields have a very basic structure.
```
<label class="dc-label" for="requiredInput">
    Required input
    <span class="dc-label__sub">required</span>
</label>
<input class="dc-input" type="text" id="requiredInput" placeholder="e. g. give an example" />
```
screen shot 2017-10-12 at 10 36 05

This simple structure breaks as soon as an error text is added.
```
<label class="dc-label" for="errorInput">
    Required input – Error
    <span class="dc-label__sub dc-label__sub--is-error">required</span>
</label>
<input class="dc-input dc-input--is-error" type="text" id="errorInput" placeholder="e. g. give an example" />
<span class="dc--text-error">Please don't leave this empty.</span>
```
screen shot 2017-10-12 at 10 36 11

The error text should be placed below the input without breaking vertical rhythm and spacing. 
It seems a parent element is necessary to deal with different permutations of input fields and as a way of adding inline and block form element support. See Bootstrap [stacked](https://getbootstrap.com/docs/4.0/components/forms/#default-stacked) and [inline](https://getbootstrap.com/docs/4.0/components/forms/#inline) for reference.

### PR DESCRIPTION
**Pull request by rbarilani**
Add input-stack block/molecule to close #396, this approach was taken to don't introduce breaking changes in v2. Must be reconsidered in v3.

closes #396

@xonic @sambhav-gore I am not 100% sure about this feature

### CODE REVIEW

**Comment by sambhav-gore** [comment]
@rbarilani  I like the idea of having a stacked input for keeping vertical rythm, but would like to suggest a change in the implementation slightly. 
It looks good if feedback message is displaying all the time, but if the message is conditionally shown (like an error validation), then at the time when the message is not shown the margins are reset and whole page kind-of `moves`.

My suggestion would be to have the feedback placed through absolute positioning OR the margins below the fields are adjusted accordingly when the message Is present or is not present.

In simple terms, other fields on the same page should not move when there is an error/help message on one field.

**Comment by rbarilani** [comment]
@sambhav-gore Yeah.. I thought about that, on the other hand with absolute positioning you can fall in never-ending problem with long messages and responsive/fluid blocks.

I took inspiration from [Bootstrap](http://getbootstrap.com/docs/4.0/components/forms/#custom-styles)

**Comment by sambhav-gore** [comment]
@rbarilani that is correct as well. How about giving a modifier for absolute positioned one? those who have smaller (one-liner) messages could go for absolutely placed alternative Or would it be an overkill for this as it is something simple users can implement as well?

**Comment by rbarilani** [comment]
@sambhav-gore IHMO it's overkill. I would like to re-think this in v3.

**Comment by xonic** [comment]
I agree we should first keep it simple and think about a better solution in v3 that also prevents form bouncing. Maybe a clever design might help with bouncing, overflows, multiple lines, etc.
Need to think about it.

**Comment by rbarilani** [comment]
👍

**Comment by sambhav-gore** [comment]
👍
